### ISSUE TITLE
Function playSpotifyRadio(artistId, artistName) generates UPnP error 701

### ISSUE THREAD

**Issue by magnlund** [bencevans/node-sonos#392]
When trying to play Spotify Radio using function playSpotifyRadio(artistId, artistName) error UPnP 701 is raised. It used to work before but it does not anylonger. I use the code from the Example and have also verified correct Region settings (EU). All of the other Spotify play functions work fine though. Please advice

## Versions (and Environment)
**Node version**: 8.9.3
**node-sonos version**: 1.6.0
**OS**: Linux (RPi)

**Comment by Villarrealized** [comment]
I can confirm that this issue exists in v1.8.1. 

It looks like Spotify has changed some the query flags and parameters needed when playing an artistRadio Id.

Specifically, `sid=9` has changed to `sid=12` and `sn=7` changed to `sn=5`

```javascript
// This fails with UPnP error 701
device.playSpotifyRadio('5schNIzWdI9gJ1QRK8SBnc','Ben Howard')
// This also fails, which is essentialy the same thing
device.setAVTransportURI({ uri: "x-sonosapi-radio:spotify:artistRadio:5schNIzWdI9gJ1QRK8SBnc?sid=9&amp;flags=8300&amp;sn=7", metadata: "" })

// Both of these work

// Play Ben Howard radio
device.setAVTransportURI({ uri: "x-sonosapi-radio:spotify:artistRadio:5schNIzWdI9gJ1QRK8SBnc?sid=12&amp;flags=8300&amp;sn=5", metadata: "" })

// Play Sarah Sparks radio
device.setAVTransportURI({ uri: "x-sonosapi-radio:spotify:artistRadio:1rNB96O30Md5JMUNgWwg99?sid=12&amp;flags=8300&amp;sn=5", metadata: "" })
```

We would need to change the following line in helpers.js

```javascript
// Replace this line
uri: 'x-sonosapi-radio:' + spotifyUri + '?sid=9&flags=8300&sn=7',

// With this line
uri: 'x-sonosapi-radio:' + spotifyUri + '?sid=12&flags=8300&sn=5',
```

**Comment by magnlund** [comment]
Tried to change in helpers.js according to above solution but still the same issue

**Comment by Villarrealized** [comment]
Did you try this?

```javascript
// Play Ben Howard radio
device.setAVTransportURI({ uri: "x-sonosapi-radio:spotify:artistRadio:5schNIzWdI9gJ1QRK8SBnc?sid=12&amp;flags=8300&amp;sn=5", metadata: "" })
```

**Comment by magnlund** [comment]
Tried that now instead of using the Playradio function and still the same issue

Regards
Magnus Lundberg
__________________________________
Magnus Lundberg
Phone: +46 730 23 09 36
Email: contact.magnus.lundberg@protonmail.com

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by magnlund** [comment]
Could it be a difference in the parameters depending on US or EU region?

Regards
Magnus Lundberg
__________________________________
Magnus Lundberg
Phone: +46 730 23 09 36
Email: contact.magnus.lundberg@protonmail.com

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by svrooij** [comment]
@Villarrealized Spotify be like..... hate them for differentiating between regions. Maybe we should tackle this with queries based on region.

**Comment by magnlund** [comment]
Upgraded to 1.8.1 and changed the parameters and also tested with the avTransport function directly. Same issue! I'm located in EU, Sweden

Regards
Magnus Lundberg
__________________________________
Magnus Lundberg
Phone: +46 730 23 09 36
Email: contact.magnus.lundberg@protonmail.com

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by magnlund** [comment]
Any suggestions on valid parameters for EU?

Regards
Magnus Lundberg
__________________________________
Magnus Lundberg
Phone: +46 730 23 09 36
Email: contact.magnus.lundberg@protonmail.com

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by svrooij** [comment]
The way I always find the correct commands is by sniffing out the data from the desktop application with Wireshark. (Sniff WiFi or ether net card, set the filter to tcp port 1400, send the command from the desktop app, extract the soap commands for wireshark)

 Will try tomorrow, and will setup a page on the docs on how to do this.

**Comment by Villarrealized** [comment]
Interesting. I changed the spotify region back and forth from US to EU and ran the same commands multiple times and I had no issue.

@magnlund Are you sure that you have an active Spotify Premium account connected to your Sonos?

**Comment by magnlund** [comment]
I have verified that my premium account is linked to my Sonos and played around with the parameters in different ways. Issue still remains. What makes it weird is that it USED to work I think it was unitil August last year and then suddenly stopped working. I use the majority of the URIs on Spotify with Node Sonos and all of them work correct except this. Tried to reauthorize the account and verified premium and still same issue. Right now running out of ideas since it seems to be working on your side. Will await next version from you.
